testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -283,12 +281,55 @@
   page.execute_script("ace.edit('contents-editor').insert('#{arg1}')")
 end
 
-When(/^I click System List, under Systems node$/) do
-  find(:xpath, "//div[@id='nav']/nav/ul/li[contains(@class, 'active')
-       and contains(@class, 'open')
-       and contains(@class,'node')]/ul/li/div/a[contains(.,'System List')]").click
+#
+# Menu links
```

Comment:
I didn't tested, but I would simply the code with something like that:

´´´
When(/^I follow "([^"]*)" in the left menu$/) do |menu_path|
  menu_steps = menu_path.split('>').map(&:strip)
  menu_steps.each do |step|
    within(:xpath, '//aside') do
      step_parent = "#{step}/parent::div/parent::li"
      unless all(:xpath, step_parent).any? and find(:xpath, step_parent)[:class].include?('open')
        click_link(step, wait: CLICK_TIMEOUT, match: :first)
      end
    end
  end
end
´´´

Reply:
@srbarrios I refactored the steps into one dynamic only. I reuse some of your logic but still I need to keep the `find` and `click` on parent when the submenu needs to be open. You can take a look [here](https://github.com/uyuni-project/uyuni/pull/937/commits/24b0b63404054089ab499792fea0328592e01e7c#diff-69eaaf5982104837a8e6521c7fa3d4feR295)